akka-bench-jmh/src/main/scala/akka/stream/FusedGraphsBenchmark.scala

```diff
@@ -0,0 +1,322 @@
+    }
+
+    (logic, latch)
+  }
+}
+
+class IdentityStage extends GraphStage[FlowShape[MutableElement, MutableElement]] {
+  val in = Inlet[MutableElement]("Identity.in")
+  val out = Outlet[MutableElement]("Identity.out")
+  override val shape = FlowShape(in, out)
+
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) with InHandler with OutHandler {
+    override def onPush(): Unit = push(out, grab(in))
+    override def onPull(): Unit = pull(in)
+
+    setHandlers(in, out, this)
+  }
+}
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Array(Mode.Throughput))
+class FusedGraphsBenchmark {
+  import FusedGraphsBenchmark._
+
+  implicit val system = ActorSystem("test")
+  var materializer: ActorMaterializer = _
+  var testElements: Array[MutableElement] = _
+
+  var singleIdentity: RunnableGraph[CountDownLatch] = _
+  var chainOfIdentities: RunnableGraph[CountDownLatch] = _
+  var singleMap: RunnableGraph[CountDownLatch] = _
+  var chainOfMaps: RunnableGraph[CountDownLatch] = _
+  var repeatTakeMapAndFold: RunnableGraph[CountDownLatch] = _
+  var singleBuffer: RunnableGraph[CountDownLatch] = _
+  var chainOfBuffers: RunnableGraph[CountDownLatch] = _
+  var broadcastZip: RunnableGraph[CountDownLatch] = _
+  var balanceMerge: RunnableGraph[CountDownLatch] = _
+  var broadcastZipBalanceMerge: RunnableGraph[CountDownLatch] = _
+
+  @Setup
+  def setup(): Unit = {
+    val settings = ActorMaterializerSettings(system)
+      .withFuzzing(false)
+      .withSyncProcessingLimit(Int.MaxValue)
+      .withAutoFusing(false) // We fuse manually in this test in the setup
+
+    materializer = ActorMaterializer(settings)
+    testElements = Array.fill(ElementCount)(new MutableElement(0))
+    val addFunc = (x: MutableElement) => { x.value += 1; x }
+
+    val testSource = Source.fromGraph(new TestSource(testElements))
+    val testSink = Sink.fromGraph(new JitSafeCompletionLatch)
+
+    def fuse(r: RunnableGraph[CountDownLatch]): RunnableGraph[CountDownLatch] = {
+      RunnableGraph.fromGraph(Fusing.aggressive(r))
+    }
+
+    val identityStage = new IdentityStage
+
+    singleIdentity =
+      fuse(
+        testSource
+          .via(identityStage)
+          .toMat(testSink)(Keep.right)
+      )
+
+    chainOfIdentities =
+      fuse(
+        testSource
+          .via(identityStage)
+          .via(identityStage)
+          .via(identityStage)
+          .via(identityStage)
+          .via(identityStage)
+          .via(identityStage)
+          .via(identityStage)
+          .via(identityStage)
+          .via(identityStage)
+          .via(identityStage)
+          .toMat(testSink)(Keep.right)
+      )
+
+    singleMap =
+      fuse(
+        testSource
+          .map(addFunc)
+          .toMat(testSink)(Keep.right)
+      )
+
+    chainOfMaps =
+      fuse(
+        testSource
+          .map(addFunc)
+          .toMat(testSink)(Keep.right)
+      )
```

Comment:
interesting, this have been fixed, but GH is confused and shows the old one here instead of hiding the comment.